common/compositor/va/varenderer.cpp

```diff
@@ -169,39 +169,42 @@ bool VARenderer::GetVAProcDeinterlaceFlagFromVideo(HWCDeinterlaceFlag flag) {
 }
 bool VARenderer::SetVAProcFilterDeinterlaceMode(
     const HWCDeinterlaceProp& prop) {
+  VAProcDeinterlacingType mode;
   if (prop.flag_ == HWCDeinterlaceFlag::kDeinterlaceFlagNone) {
     SetVAProcFilterDeinterlaceDefaultMode();
     return true;
   } else if (prop.flag_ == HWCDeinterlaceFlag::kDeinterlaceFlagForce ||
              GetVAProcDeinterlaceFlagFromVideo(prop.flag_)) {
     switch (prop.mode_) {
       case HWCDeinterlaceControl::kDeinterlaceNone:
-        deinterlace_caps_.mode_ = VAProcDeinterlacingNone;
+        mode = VAProcDeinterlacingNone;
         break;
       case HWCDeinterlaceControl::kDeinterlaceBob:
-        deinterlace_caps_.mode_ = VAProcDeinterlacingBob;
+        mode = VAProcDeinterlacingBob;
         break;
       case HWCDeinterlaceControl::kDeinterlaceWeave:
-        deinterlace_caps_.mode_ = VAProcDeinterlacingWeave;
+        mode = VAProcDeinterlacingWeave;
         break;
       case HWCDeinterlaceControl::kDeinterlaceMotionAdaptive:
-        deinterlace_caps_.mode_ = VAProcDeinterlacingMotionAdaptive;
+        mode = VAProcDeinterlacingMotionAdaptive;
         break;
       case HWCDeinterlaceControl::kDeinterlaceMotionCompensated:
-        deinterlace_caps_.mode_ = VAProcDeinterlacingMotionCompensated;
+        mode = VAProcDeinterlacingMotionCompensated;
         break;
       default:
         ETRACE("Hwc unsupport deinterlace mode\n");
         return false;
     }
-  for (int i = 0; i < VAProcDeinterlacingCount; i++) {
-    if (deinterlace_caps_.caps_[i].type == deinterlace_caps_.mode_) {
-      update_caps_ = true;
-      return true;
+    for (int i = 0; i < VAProcDeinterlacingCount; i++) {
+      if (deinterlace_caps_.caps_[i].type == mode &&
+          deinterlace_caps_.mode_ != mode) {
+        deinterlace_caps_.mode_ = mode;
+        update_caps_ = true;
+        return true;
+      }
     }
-  }
-  ETRACE("VA Filter unsupport deinterlace mode\n");
-  return false;
+    ETRACE("VA Filter unsupport deinterlace mode\n");
```

Comment:
hmm if  deinterlace caps hasn't changed, we will print this error ?

Reply:
@kalyankondapally , deinterlace_caps_.caps_[i] saves the supported deinterlace mode queried from LIBVA driver. Before we set a deinterlace mode to driver, we need to know if the mode supported by LIBVA driver, and if not, here will give out a message that LIBVA driver does not support this deinterlace mode.